### PR TITLE
Mention limited dynamic linking

### PR DESCRIPTION
**Pull request by jfbastien**
Further clarifications will be needed to explain the Wasm object's API. That should be done in a separate PR.

### CODE REVIEW

File: DynamicLinking.md [hunk 1]
```diff
@@ -2,10 +2,13 @@
 
 Dynamic loading of code is in [the MVP](MVP.md) in the form of
 [modules](Modules.md), but all loaded modules have their own separate
-[linear memory](AstSemantics.md#linear-memory) and cannot share
-[function pointers](AstSemantics.md#calls). Dynamic linking will allow
-developers to share memory and function pointers between WebAssembly
-dynamic libraries.
+[linear memory](AstSemantics.md#linear-memory) by default and cannot share
+[function pointers](AstSemantics.md#calls). Limited collaboration between
+modules is possible im the MVP by having two modules share the same linear
```
Comment (lukewagner) [inline DynamicLinking.md]: s/im/in/

Comment (jfbastien) [inline DynamicLinking.md]: Oops, done.

File: DynamicLinking.md [hunk 2]
```diff
@@ -2,10 +2,13 @@
...
 Dynamic loading of code is in [the MVP](MVP.md) in the form of
 [modules](Modules.md), but all loaded modules have their own separate
-[linear memory](AstSemantics.md#linear-memory) and cannot share
-[function pointers](AstSemantics.md#calls). Dynamic linking will allow
-developers to share memory and function pointers between WebAssembly
-dynamic libraries.
+[linear memory](AstSemantics.md#linear-memory) by default and cannot share
+[function pointers](AstSemantics.md#calls). Limited collaboration between
+modules is possible in the MVP by having two modules share the same linear
+memory and invoke each other through the embedder.
```
Comment (lukewagner) [inline DynamicLinking.md]: s/embedder/host environment/

File: DynamicLinking.md [hunk 3]
```diff
@@ -2,10 +2,13 @@
...
+
+True dynamic linking will allow developers to share memory *and* function
+pointers between WebAssembly dynamic libraries.
```
Comment (lukewagner) [inline DynamicLinking.md]: Perhaps "... share memory, function pointers, and future non-memory state like global variables and thread-local variables."

**Comment by lukewagner** [comment]
Agreed on writing up the full Wasm JS API in a separate PR.  (FWIW, I have that as a todo on my list after switching to postorder.)

**Comment by lukewagner** [comment]
lgtm with these changes

**Comment by jfbastien** [comment]
Yup, no rush: I just wanted to get the basic clarification out there!

**Comment by sunfishcode** [comment]
I see this was rushed in before I could comment ;), however I'm ok with this, having recently had some conversations about how we may actually use this general approach as the first step to actual dynamic linking. My main request would be to emphasize that there is no stable ABI implied yet!

**Pull request merged**